internal/creds.go

```diff
@@ -99,7 +99,7 @@ func credentialsFromJSON(ctx context.Context, data []byte, ds *DialSettings) (*g
 }
 
 func isSelfSignedJWTFlow(data []byte, ds *DialSettings) (bool, error) {
-	if (ds.EnableJwtWithScope || ds.HasCustomAudience() || len(ds.GetScopes()) == 0) &&
+	if ((ds.EnableJwtWithScope && len(ds.GetScopes()) > 0) || ds.HasCustomAudience()) &&
```

Comment:
I don't feel we need to check for the scope here when scope is enabled for JWT

1) If scope is non-empty, JWT is used
2) If scope is empty, audience must be non-empty, in this case JWT is used too.

Does this make sense?

Reply:
I agree, will remove as that is another way to fix this.